vitrina/smart_contracts/forms.py

```diff
@@ -50,3 +57,33 @@ class SmartContractFormSetHelper(FormHelper):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.form_tag = False
+
+
+class AgreementUploadForm(forms.ModelForm):
+    class Meta:
+        model = AgreementFile
+        fields = ("file",)
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["file"].validators = [
+            FileExtensionValidator(
+                allowed_extensions=["adoc"],
+                message=_("Dokumentas turi būti adoc formato."),
+            )
+        ]
+        self.helper = FormHelper()
+        self.helper.add_input(
+            Submit("submit", _("Įkelti"), css_class="button is-primary")
+        )
+
+    def clean_file(self) -> FieldFile:
+        file = self.cleaned_data["file"]
+        try:
+            # TODO: check for multiple signatures, if AgreementStatuses.INITIATED
+            # TODO: check agreement checksum
```

Comment:
Will this be done with an upcoming task or the same one? If an upcoming one, let's add the ticket as well.